src/app/(auth)/signup/page.tsx

```diff
@@ -14,8 +14,28 @@ export default function SignUp() {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [emailSentCount, setEmailSentCount] = useState(0);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
   // const { push } = useRouter();
   const handleSignUp = async () => {
+    if (email === '' || password === '') {
+      if (email === '') {
+        setEmailError('Invalid Email');
+      } else {
+        setEmailError('');
+      }
+      if (password === '') {
+        setPasswordError('Invalid Password');
+      } else {
+        setPasswordError('');
+      }
+      setErrorMessage('');
+      return;
+    }
```

Comment:
we can probably unwrap the bigger if statement and add it here
i.e. `if (email === '' || password === '') return;`